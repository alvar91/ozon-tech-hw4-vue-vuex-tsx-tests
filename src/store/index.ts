import Vue from "vue";
import Vuex from 'vuex';
import { createVuexStore } from "vuex-simple";

// @ts-ignore
import { MainModule } from "./modules/main/main.ts";

Vue.use(Vuex);

const store = createVuexStore(new MainModule());

export default store;
