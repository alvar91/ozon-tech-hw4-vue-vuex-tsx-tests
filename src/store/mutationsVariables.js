export const FETCH_USERS_ACTION = "FETCH_USERS_ACTION";
export const FETCH_USERS = "FETCH_USERS";
export const SET_LOADING = "SET_LOADING";
export const SET_CURRENT_USER = "SET_CURRENT_USER";
export const SET_FILTER_KEYWORD = "SET_FILTER_KEYWORD";
export const SET_FORM_INPUT = "SET_FORM_INPUT";
export const SET_FORM_CHANGE = "SET_FORM_CHANGE";
export const SET_FORM_RADIO = "SET_FORM_RADIO";
