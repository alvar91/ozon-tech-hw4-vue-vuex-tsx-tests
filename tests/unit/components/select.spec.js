import Vuex from "vuex";
import { mount, createLocalVue } from "@vue/test-utils";

import Select from "@/components/select.vue";

import { mockUserName, mockUserId, mockUser } from "../../mock/mockUser";

import {
  FETCH_USERS_ACTION,
  SET_CURRENT_USER,
} from "@/store/mutationsVariables";

import store from "@/store/index";

const localVue = createLocalVue();
localVue.use(Vuex);

const { citizenship } = mockUser;
describe("Select.vue", () => {
  let wrapper;

  beforeEach(async () => {
    await store.dispatch(FETCH_USERS_ACTION);
    store.commit(SET_CURRENT_USER, mockUserId);
  });

  it("should correctly renders markup with default props", () => {
    wrapper = mount(Select, {
      localVue,
      propsData: {
        title: "Гражданство",
        name: "citizenship",
        value: { citizenship },
      },
      store,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should correctly select data and change the store", () => {
    wrapper = mount(Select, {
      localVue,
      propsData: {
        title: "Гражданство",
        name: "citizenship",
        value: { citizenship },
      },
      store,
    });

    const oldCitizenship = store.getters.currentUser.citizenship;

    const select = wrapper.find("select");
    const options = select.findAll("option");

    let count = 0;
    while (options.at(count).text() === citizenship) {
      count++;
    }

    const newCitizenOption = options.at(count);
    select.element.value = newCitizenOption.text();
    select.trigger("change");

    expect(store.getters.currentUser.citizenship).not.toBe(oldCitizenship);
  });

  afterEach(() => {
    wrapper.destroy();
  });
});
