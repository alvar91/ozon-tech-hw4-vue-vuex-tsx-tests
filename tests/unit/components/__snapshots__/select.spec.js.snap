// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select.vue should correctly renders markup with default props 1`] = `
<label class="card__field"><span class="card__title">Гражданство</span><select name="citizenship" class="card__select card__select--arrow">
    <option value="Грузия">Грузия</option>
    <option value="США">США</option>
    <option value="Российская Федерация">Российская Федерация</option>
    <option value="Украина">Украина</option>
    <option value="Беларусь">Беларусь</option>
    <option value="Казахстан">Казахстан</option>
  </select></label>
`;
